images/macos/provision/configuration/CreateCleanAnkaTemplate.ps1

```diff
@@ -0,0 +1,232 @@
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [Version] $MacOSVersion,
+    [ValidateNotNullOrEmpty()]
+    [String] $TemplateUsername,
+    [Parameter(Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [String] $TemplatePassword,
+    [ValidateNotNullOrEmpty()]
+    [String] $RegistryUrl,
+    [Bool] $BetaSearch = $false,
+    [Int] $CpuCount = 6,
+    [Int] $RamSizeGb = 7,
+    [Int] $DiskSizeGb = 300,
+    [String] $DisplayResolution = "1920x1080"
+)
+
+$ErrorActionPreference = "Stop"
+
+function Get-MacOSInstaller {
+    param (
+        [Version] $MacOSVersion,
+        [Bool] $BetaSearch = $false
+    )
+
+    # Enroll machine to DeveloperSeed if we need beta and unenroll otherwise
+    if ($BetaSearch)
+    {
+        Write-Host "Beta Version requested. Enrolling machine to DeveloperSeed"
+        sudo /System/Library/PrivateFrameworks/Seeding.framework/Versions/Current/Resources/seedutil enroll DeveloperSeed | Out-Null
+    }
+    else
+    {
+        Write-Host "Reseting the seed before requesting stable versions"
+        sudo /System/Library/PrivateFrameworks/Seeding.framework/Versions/Current/Resources/seedutil unenroll | Out-Null
+    }
+
+    $availableVersions = Get-AvailableVersions -IsBeta $BetaSearch
+
+    if (-not ($MacOSVersion -in $availableVersions.OsVersion))
+    {
+        Write-Host "Requested macOS version '$MacOSVersion' not found in the list of avaialable installers. Avaialable versions are:`n$($availableVersions.OsVersion)"
+        Write-Host 'Make sure to pass "-BetaSearch $true" if you need a beta version installer'
+        exit 1
+    }
+
+    # Validate there is no softwareupdate at the moment
+    Test-SoftwareUpdate
+
+    $macOsName = ($availableVersions | Where-Object { $_.OsVersion -match $MacOSVersion }).OsName
+    $installerPathPattern = "/Applications/Install*${macOsName}.app"
+
+    if (Test-Path $installerPathPattern) {
+        Write-Host "Removing existed macOS $macOsName installation package before downloading the new one"
+        bash -c "sudo rm -rf $installerPathPattern"
+    }
+
+    Write-Host "Requested macOS version '$MacOSVersion' installer found, fetching it from Apple"
+    softwareupdate --fetch-full-installer --full-installer-version $MacOSVersion
```

Comment:
Should we add multiple attempts to try download this file?

Reply:
Good idea, we can implement retry logic here